xrtl/base/geometry.h

```diff
@@ -33,6 +33,27 @@ struct Point2D {
   bool operator!=(const Point2D& other) const { return !(*this == other); }
 };
 
+struct Point3D {
+  int x = 0;
```

Comment:
not meant for 3d primitives, but for offsets/dimensionality of things like windows, textures, framebuffers, etc. we'll want to use glm and such for any real 3d primitives.